terentwc iP

Terentwc opened the pull request:
# Krunch

> __"I want their captain, Peter Pan."__ - Captain Hook

## Chatbot companion [that will never gonna give you up](https://www.youtube.com/watch?v=dQw4w9WgXcQ) 😁

### Never gonna
* Let you down
* Run around and desert you

### Things  to do:
- [ ] **Peter Pan**
- [x] Sulk in Neverland<sup>TM</sup>
- [ ] Capture Tinkle Bell

### Numbered List:
1. ~~me~~ Captain Hook
2. Mr. Smee
3. Parrot

### Greetings person from the internet
```
public void greet() {
    System.out.println("Oh, what a treat. Finally, you've graced me with your presence.");
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Terentwc committed:
Level 0 Renaming, Greeting/Asking/Farewell

Terentwc committed:
Level 1 Personality change, Echo

Terentwc committed:
Level 2 Add, List

Terentwc committed:
Level 3 Mark/Unmark and A class

Terentwc committed:
Level 4 todo,event,deadline

Terentwc committed:
Automated text ui test

It sort of works but converting to UTF8 sucks. even with forcing intellij to use utf8 or whatever i found in google. the second run with the same exact code gives an error that says UTF8 is the wrong encoding type. (might come back later or smth to do this or not but it's sucking up so much of my time)

Terentwc committed:
Level 5 Errors and Exception

Terentwc committed:
Level 6 Delete

Terentwc committed:
Enum

Added Enum for possible future use

xenyeo commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task{
```
```suggestion
public class Deadline extends Task {
```
Whitespace should be followed after class name.

xenyeo replied:
Also missing header comment for public class. If not LGTM, just some nits to fix.

xenyeo commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,14 @@
+public class Event extends Task {
```
Missing header comment for public class, if not LGTM.

xenyeo commented on src/main/java/IllegalException.java:
```diff
@@ -0,0 +1,5 @@
+    public IllegalException(String e) {
+        super(e);
+    }
+}
```
LGTM. A suggestions is that the name of the exception can be more specific, eg. KrunchException.

xenyeo commented on src/main/java/Krunch.java:
```diff
@@ -0,0 +1,221 @@
+import java.util.Scanner;
+import java.util.ArrayList;
```
It might be a good practice to organize imports from the same package group alphabetically.

xenyeo commented on src/main/java/Krunch.java:
```diff
@@ -0,0 +1,221 @@
+        }
+
+    }
+}
```
Missing header comment for class.
Code logic can be shortened by introducing more classes to make it more OOP.

xenyeo commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,35 @@
+public class Task {
+    protected String task;
+    protected boolean done;
```
```suggestion
    protected boolean isDone;
```
This boolean variable should be named to sound like booleans.

xenyeo commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,35 @@
+    // initialing task as not done
+    public Task (String task) {
+        this.task = task;
+        this.done = false;
```
```suggestion
        this.isDone = false;
```

xenyeo commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    public String stringDone() {
+        return done ? "X" : " ";
```
```suggestion
        return isDone ? "X" : " ";
```

xenyeo commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,35 @@
+    public String toString() {
+        return "[" + stringDone() + "] " + getTask();
+    }
+}
```
Missing header comment for class.
Header comments should be added for public methods instead of normal comments.

xenyeo commented on src/main/java/TaskAction.java:
```diff
@@ -0,0 +1,12 @@
+    BYE,
+    INVALID,
+    BLANK
+}
```
Missing header comment for public enum, if not LGTM.

xenyeo commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,11 @@
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+}
```
Missing header comment, if not LGTM.

xenyeo reviewed (commented):
Week 3 iP tasks not done.

glennliew commented on src/main/java/Krunch.java:
```diff
@@ -0,0 +1,221 @@
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        while(true) {
+            String UserInput = scanner.nextLine();
```
Not in camelCase
```suggestion
            String userInput = scanner.nextLine();
```

glennliew commented on src/main/java/Krunch.java:
```diff
@@ -0,0 +1,221 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
+public class Krunch {
```
Missing header comments

Terentwc committed:
branch level 7

saving to tasks.txt first

Terentwc committed:
branch level 7 2

Updated to save and load from file

Terentwc committed:
Level 8 Fixed Date

Deadline and Event now needs to follow yyyy-mm-dd. will make it be more flexible next time

glennliew commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,23 @@
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+    protected LocalDate by;
```
variable name by is short and vague. Perhaps could have a more descriptive variable name such as deadlineDate

glennliew reviewed (commented):
Some naming issues to be fixed but other than that good job!

Terentwc committed:
Partial TaskManager, Packaged Class properly

TaskManager cannot be called yet

Terentwc committed:
Updated main to be more oop

Terentwc committed:
Following coding standards before packaging

Terentwc committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

Terentwc committed:
implement gradle

Terentwc committed:
Create two tests junit

Terentwc committed:
Edit file path for main class

Terentwc committed:
add Find

Terentwc committed:
add Javadocs

Terentwc committed:
Coding style changed to follow guide

Terentwc committed:
add checkstyle.xml and supressions.xml

Terentwc committed:
add GUI into project

Terentwc committed:
integrate list to gui

Terentwc committed:
refactor the rest of the commands to be compatible with GUI

Terentwc committed:
update the GUI

Terentwc committed:
Add assertions to validate assumptions in Parser

Assertions are added to ensure that certain conditions hold at key points in the Parser class. These assertions help catch programming errors early by enforcing assumptions about input validation and method behavior.

Assertions added:
- Ensure `UserInput` is never null before parsing.

These assertions serve as internal correctness checks and are not a replacement for exception handling, which still manages user input validation.

Terentwc committed:
Refactor Code to fit code quality

Refactor code for improved readability and efficiency

- Reduced nested logic with early returns
- Simplified exception handling for cleaner code flow
- Optimized repeated method calls to improve performance
- Extracted common strings and logic for better maintainability
- Enhanced task-related methods for clarity and consistency

Terentwc committed:
Introduce a better find function

Terentwc committed:
Update README.md

Add user guide as README.md

Terentwc committed:
change user and Krunch icons

sprinkle flavor in Krunch by changing the icon

Terentwc committed:
Add UI.png

Things changed:
- Add UI.png to docs
- Add Krunch as title of the GUI window

Terentwc committed:
Update README

Add in Acknowledgements
